### PR TITLE
Unittest interface

### PR DESCRIPTION
**Pull request by CunliangGeng**
- fix bugs in interface.py
- add thorough unittests for interface.py functions

Note: please first check and merge PR #11, then this one.

### CODE REVIEW

File: pdb2sql/StructureSimilarity.py [hunk 1]
```diff
@@ -311,15 +310,20 @@ def compute_izone(self,cutoff=5.0,save_file=True,filename=None):
         Returns:
             dict: i-zone definition
         """
+        backbone = ['CA', 'C', 'N', 'O']
         sql_ref = interface(self.ref)
-        contact_ref = sql_ref.get_contact_atoms(cutoff=cutoff,extend_to_residue=True,return_only_backbone_atoms=True)
+        contact_ref = sql_ref.get_contact_atoms(cutoff=cutoff,extend_to_residue=True)
```
Comment (NicoRenaud) [inline pdb2sql/StructureSimilarity.py]: why did you change the return only backbone ? In my experience it needs to be exactly like that to obtain the same results as PROFIT

File: pdb2sql/StructureSimilarity.py [hunk 2]
```diff
@@ -658,6 +662,8 @@ def compute_irmsd_pdb2sql(self,cutoff=10,method='svd',izone=None,exportpath=None
         Returns:
             float: i-RMSD value of the conformation
         """
+        backbone = ['CA', 'C', 'N', 'O']
```
Comment (NicoRenaud) [inline pdb2sql/StructureSimilarity.py]: should we make that into a class variable ? so that we do not have to redefine it each time

File: pdb2sql/StructureSimilarity.py [hunk 3]
```diff
@@ -666,9 +672,12 @@ def compute_irmsd_pdb2sql(self,cutoff=10,method='svd',izone=None,exportpath=None
         if izone is None:
             contact_ref = sql_ref.get_contact_atoms(
                 cutoff=cutoff,
-                extend_to_residue=True,
-                return_only_backbone_atoms=True)
-            index_contact_ref = contact_ref['A'] + contact_ref['B']
+                extend_to_residue=True)
+            index_contact_ref = []
+            for v in contact_ref.values():
```
Comment (NicoRenaud) [inline pdb2sql/StructureSimilarity.py]: Is this for multi chain support ? why did you change here as well the only_backbone_atom ?

File: pdb2sql/interface.py [hunk 4]
```diff
@@ -29,14 +29,32 @@ def get_contact_atoms(
             extend_to_residue=False,
             only_backbone_atoms=False,
             excludeH=False,
```
Comment (NicoRenaud) [inline pdb2sql/interface.py]: there was a distinction here as to consider only the backbone atoms and return only the backbone atoms. I'm not sure that's correct here anymore. I have a test somewhere where I compare the results with PROFIT. We should make that part of the testing as well

File: pdb2sql/pdb2sqlcore.py [hunk 5]
```diff
@@ -228,63 +242,72 @@ def _fix_chainID(self):
...
+            for i in columns.split(','):
+                if i.strip() not in valid_colnames:
+                    raise ValueError(
+                        f'Invalid column name {i}. Possible names are\n'
+                        f'{self.get_colnames()}')
 
         # the asked keys
         keys = kwargs.keys()
 
-        # check if the column exists
```
Comment (NicoRenaud) [inline pdb2sql/pdb2sqlcore.py]: why did you remove that?

File: pdb2sql/pdb2sqlcore.py [hunk 6]
```diff
@@ -410,54 +433,58 @@ def get(self, atnames, **kwargs):
...
+            for i in columns.split(','):
+                if i not in valid_colnames:
+                    raise ValueError(
+                        f'Invalid column name {i}. Possible names are\n'
+                        f'{self.get_colnames()}')
 
         # the asked keys
         keys = kwargs.keys()
 
-        # check if the column exists
```
Comment (NicoRenaud) [inline pdb2sql/pdb2sqlcore.py]: same here, why did you remove that part ?

File: pdb2sql/StructureSimilarity.py (hunk 1 above)
Comment (CunliangGeng) [reply pdb2sql/StructureSimilarity.py]: The updated code now has exactly the same results as PROFIT, which are tested in `test_StructureSimilarity.py`.

I removed the argument `return_only_backbone_atoms` from `get_contact_atoms()` is because 
1. this argument is quite confusing with the argument `only_backbone_atoms` that is to use only backbone atoms to calculate contact.
2. it is easier to get the backbone atoms by using `get()` after `get_contact_atoms()`

File: pdb2sql/StructureSimilarity.py (hunk 3 above)
Comment (CunliangGeng) [reply pdb2sql/StructureSimilarity.py]: yes, for both multi-chain support and varied chain ID support.
The argument `return_only_backbone_atom` does not exist any more  as explained above. And backbone atoms will be got by using `get()` later in the code.

File: pdb2sql/interface.py (hunk 4 above)
Comment (CunliangGeng) [reply pdb2sql/interface.py]: It's correct, the latest unit tests have included this part.

File: pdb2sql/StructureSimilarity.py (hunk 2 above)
Comment (CunliangGeng) [reply pdb2sql/StructureSimilarity.py]: Good idea! will make one.

File: pdb2sql/pdb2sqlcore.py (hunk 5 above)
Comment (CunliangGeng) [reply pdb2sql/pdb2sqlcore.py]: It is not removed but refactored. The former code queries SQL with one column per time, and the new code queries all available columns and then check the existence.

File: pdb2sql/pdb2sqlcore.py (hunk 6 above)
Comment (CunliangGeng) [reply pdb2sql/pdb2sqlcore.py]: Same as above.
